### PR TITLE
Discovery

### PR DESCRIPTION
**Pull request by eevans**
To use the daemon, each instance of Cassandra must be parameterized to supply a `-Dcassandra.instance-id=<name>` property, which is used to provide stable naming for the Graphite prefix. All locally running instances of Cassandra are discovered on startup. Metrics are collected at `{ -i, --interval }` seconds and written to the Carbon server specified by `{ -H, --carbon-host }` and `{ -p, --carbon-port }`. Internal statistics (successes and failures) are likewise written to Carbon at `{ -i, --interval }` seconds in the form of `cmcd.instances.<name>.{success,failure}`. Periodic rediscovery (required to pick up new instances, or restarts of existing instances) occurs every `{ -di, --discovery-interval }` seconds.

While it's still possible to cron this against a single instance using the TCP connection information, the main-class has changed, and so it would require a minor change to invocation to get backward compatible behavior. For this reason I've bumped the version to 2.

### CODE REVIEW

File: src/main/java/org/wikimedia/cassandra/metrics/Discovery.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    /** Grok the configured instance ID, or null if unset. */
+    private static String cassandraInstanceId(VirtualMachine vm) throws IOException {
+        return vm.getSystemProperties().getProperty(INSTANCE_PROPERTY);
+    }
+
+    public static void main(String... args) throws IOException {
+        for (Jvm j : new Discovery().getJvms().values()) {
+            System.err.println(j);
```
Comment (d00rman) [inline src/main/java/org/wikimedia/cassandra/metrics/Discovery.java]: Any specific reason to use `err` instead of `out` ?

**Comment by d00rman** [comment]
LGTM :+1: Leaving to @filippog for a final check

**Comment by filippog** [comment]
:+1: LGTM too, though the top commit says WIP (I've been trying the branch in labs and it works)

**Pull request merged**